### PR TITLE
add new controller config keys page

### LINKED ISSUE #1699: cannot find a controller config keys reference doc
Moved from https://github.com/CanonicalLtd/jujucharms.com/issues/419

I'm not finding a definitive reference for all valid config keys to pass as "--config myfile.yaml" when bootstrapping a controller. The bootstrap help text shows a few usage examples, but I'm looking for something that is documented similar to the models-config doc.

There is a handy table of all valid config gets for models:
https://jujucharms.com/docs/2.1/models-config

To be consistent, it'd be good to see a corresponding reference table for controller config, maybe even at a predictably consistent URL such as:
https://jujucharms.com/docs/2.1/controllers-config

### PR DESCRIPTION
**Pull request by matthewhelmke**
Current commit is a first draft and is missing info. Eventually, this will fix #1699.

@mitechie who is the best person to ask for assistance filling in the missing details?

### CODE REVIEW

**Comment by matthewhelmke** [comment]
@howbazaar, it will very likely be faster and easier for you to look at this PR and comment than for us to meet in a hangout.

**Comment by matthewhelmke** [comment]
Gentle nudge, @howbazaar . If possible, we would really love to get this reviewed so we can push it.

File: src/en/controllers-config.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+juju controller-config
+```
+
+This will include all the currently set key values - whether they were set
+by you, inherited as a default value or dynamically set by Juju. 
+
+A key's value may be set for the current model using the same command:
+
+```bash
+juju controller-config auditing-enabled=true
```
Comment (howbazaar) [inline src/en/controllers-config.md]: Probably worth noting that many of the settings are read only once the controller has been bootstrapped.

File: src/en/controllers-config.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+|:---------------------------|--------|----------|--------------------------|:---------|
+api-port                     | integer | 17070    |                          | The port to use for connecting to the API
+auditing-enabled             | bool   | false    | false/true               | Sets whether the controller will record auditing information
+autocert-dns-name            | string |          |                          | Sets the DNS name of the controller. If a client connects to this name, an official certificate will be automatically requested. Connecting to any other host name will use the usual self-generated certificate.
+autocert-url                 | string |          |                          | Sets the URL used to obtain official TLS certificates when a client connects to the API. By default, certificates are obtained from LetsEncrypt. A good value for testing is "https://acme-staging.api.letsencrypt.org/directory".
+allow-model-access           | bool   |          |                          | Sets whether the controller will allow users to connect to models they have been authorized for even when they don't have any access rights tot he controller itself.
```
Comment (howbazaar) [inline src/en/controllers-config.md]: s/tot he/to the/

File: src/en/controllers-config.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ca-cert                      | string |          |                          | The certificate of the CA that signed the controller's CA certificate, in PEM format
+controller-uuid              | string |          |                          | The key for the UUID of the controller
+identity-public-key          | string |          |                          | Sets the public key of the identity manager
```
Comment (howbazaar) [inline src/en/controllers-config.md]: Is there other information on identity managers anywhere?

File: src/en/controllers-config.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+identity-url                 | string |          |                          | Sets the URL of the identity manager
+mongo-memory-profile         | string | MongoProfLow |                      | Sets whether mongo uses the least possible memory or another mongo memory profile
```
Comment (howbazaar) [inline src/en/controllers-config.md]: You should specify what the possible values are.

MongoProfLow isn't the value, the two values are "low" and "default"

File: src/en/controllers-config.md (hunk 1 above)
Comment (howbazaar) [inline src/en/controllers-config.md]: Acutally, checked the code, you can't actually set any values. Only get them.

Comment (matthewhelmke) [inline src/en/controllers-config.md]: Oh! Good to know. Thanks!

Comment (matthewhelmke) [inline src/en/controllers-config.md]: @howbazaar, so the settings can (only) be set during bootstrap? How is that done, by passing parameters in the bootstrap command?

File: src/en/controllers-config.md (hunk 3 above)
Comment (matthewhelmke) [inline src/en/controllers-config.md]: Yes, in https://jujucharms.com/docs/stable/users-models

File: src/en/controllers-config.md (hunk 4 above)
Comment (matthewhelmke) [inline src/en/controllers-config.md]: That is the area where I was hoping most for your kind assistance, @howbazaar. I don't know the valid or default values for the blank lines.

**Comment by evilnick** [comment]
@matthewhelmke controller config are set when bootstrapping with the --config option. We already have examples of it https://jujucharms.com/docs/stable/controllers-creating#change-timeout-and-retry-delays

And also, those values  ^ should be in the table!

**Comment by matthewhelmke** [comment]
Thanks @evilnick

**Comment by matthewhelmke** [comment]
Note to self: when this content is reviewed and approved, add a link back from [here](https://jujucharms.com/docs/stable/controllers-creating#change-timeout-and-retry-delays) to this new page.

**Comment by matthewhelmke** [comment]
Okay, I'll do that now, @evilnick.

**Pull request merged**